### PR TITLE
Clustering in month view

### LINKED ISSUE #1240: 🗓 Event clustering in month calendar view
## Description
A user should be able to see all events in the month calendar view, even if they can't fully fit, in which case they need to be clustered.

See epic #1233 for context.

## Relevant Job Stories
* When I have lots of events, I want to be able to see them in a calendar view, so that I can easily assess events’ durations and distribution over time
* When I view a very busy day/week of events I want to still be able to get a birds-eye view of what events will happen, so that the clutter won’t impede my event planning

## Prerequisites
Don't start this until #1235 and #1238 have been completed.

## Requirements
- [x] Display only as many event components as can fit without breaking the grid
- [x] If all events in a day can fit when rendered individually, do so
- [x] If all events can't fit, cluster:
  - [x] Cluster as multi-shift events if applicable
  - [x] Cluster as multi-location events if applicable
  - [x] Cluster all remaining events as an arbitrary cluster

## Possible implementations
Logic may have been implemented already by solutions to #1239, #1192 and #1193.

## Design specifications
Use clusters from [state matrix in Figma](https://www.figma.com/file/EjLcpjKWgNizCb5PRBF51P/Zetkin-3?node-id=4072-58606&t=mjBj8RCNKhr1F5lx-4). See the columns called "month".

image

## Open questions
None

### PR DESCRIPTION
**Pull request by richardolsson**
## Description
This PR renders events (using temporary components) in the calendar month view. Depending on how big the view is, a varying number of events will be rendered.

## Screenshots
https://github.com/zetkin/app.zetkin.org/assets/550212/6beaa3df-0919-4b10-b2be-9c4ef0fc9af2

## Changes
* Adds a new hook, `useMonthCalendarEvents()` which clusters correctly for month view
* Implements the hook and rendering of temporary components in the month view
* Adds logic to update the number of events displayed per day when the calendar resizes

## Notes to reviewer
The event components are temporary, while waiting for #1238 to finish.

## Related issues
Resolves #1240

### CODE REVIEW

**Comment by richardolsson** [comment]
Closed and reopened to re-trigger CI, which failed yesterday during a GitHub outage.

**Review by rebecarubio — APPROVED** [review]
lgtm 💯

File: src/features/calendar/components/CalendarMonthView/Day.tsx [hunk 1]
```diff
@@ -20,24 +24,45 @@ const Day = ({ date, isInFocusMonth, onClick }: DayProps) => {
 
   return (
     <Box
-      alignItems="start"
+      alignItems="stretch"
       bgcolor={isInFocusMonth ? '#eee' : 'none'}
       border="2px solid #eeeeee"
```
Comment (ziggabyte) [inline src/features/calendar/components/CalendarMonthView/Day.tsx:29]: this should be a theme color i think, but could be fixed later i guess

**Review by ziggabyte — APPROVED** [review]
Quickest sloppiest look at the code but I trust it works let's gooooooo!

**Pull request merged**